### PR TITLE
SERV-642 Merge Rest Monitoring pages

### PR DESCRIPTION
**Pull request by AlanRoth**
Merged
- asadmin-commands.adoc
- configuring-rest-monitoring
- using-rest-monitoring.adoc

### CODE REVIEW

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+----
+asadmin> change-admin-password 
+----
+
+then using the enable-secure-admin command
+
+----
+asadmin> enable-secure-admin 
+----
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:169]: This section is technical "common-knowledge" and would be present in the existing GF manuals, so I think that it is redundant

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+New users can be added using the `create-file-user` command.
+
+----
+asadmin> create-file-user --groups=rest --target=server-config --authrealmname=file ${username}
+----
+
+[[performing-read-operations]]
+=== Performing READ Operations
+
+INFO Only the READ Operation is supported.
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:205]: This sentence seems to be out of place

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+A list of attribute names can be found in the request of an empty attribute name under ‘value’. 
+
+[[bulk-read-example]]
+==== Example Execution
+
+To read the MBean `java.lang:type=Memory` using the default configuration, you would make a GET request to: `http://localhost:4848/rest-monitoring/rest/read/java.lang:type=Memory`.
+
+*Example Output*
+
+----
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:224]: `[source, json]` missing

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+  },
+  "timestamp": 1502799650273,
+  "status": 200
+}
+----
+
+[[performing-bulk-read-operations]]
+=== Performing Bulk READ Operations
+
+INFO Only the READ Operation is supported.
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:255]: Again, out-of-place sentence

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+It is possible to execute bulk operations using the REST monitoring API. To do this, issue a POST request to the REST API URL with the following JSON structure as the body payload.
+
+[[example-bulk-payload]]
+==== Example Payload
+
+The payload is a JSON array consisting of objects of type, mbean, and attribute json objects. You may send a single operation request instead of an array also.
+
+----
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:264]: `[source, json]` missing

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    "attribute" : "<ATTRIBUTE_NAME>"
+  }
+]
+----
+
+*type*: The ‘type’ of operation to execute, e.g. read, search, write, etc.
+
+*mbean*: The MBean attribute which the operation will be executed on.
+
+*attribute*: The MBean attribute upon which the operation will be executed. If omitted, all attributes of the MBean will be involved in the operation.
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:283]: This information should be put in a bullet list

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+
+[[example-bulk-execution]]
+==== Example Execution
+
+Using `curl` and a sample REST API URL of `http://localhost:4848/rest-monitoring/rest`
+
+----
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:290]: `[source, shell]` missing

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+		"mbean": "java.lang:type=Runtime",
+		"attribute" : "Uptime",
+		"type": "read"
+	}
+]
+----
+
+*Example Output*
+
+----
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:308]: `[source, json]` missing

**Review by fturizo — CHANGES_REQUESTED** [review]
Changes needed

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+
+The REST monitoring implements a Jolokia type functionality in Payara Server.
+
+It exposes MBeans over HTTP with a REST API to stop JMX monitoring requiring RMI. The REST monitoring is hosted on the admin-listener (port 4848).
+
+[[rest-monitoring-configuration]]
+== Configuring REST monitoring
+
+REST monitoring is configurable through asadmin commands. This feature is in
+technical preview, so some things may not work quite as intended.
```
[the remark this answers is not in the document]
Comment (AlanRoth) [reply docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc]: I missed this bit! It is not needed indeed

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+|Whether or not secure access to the REST Monitoring application is enabled.
+|`false`
+|no
+|====
+
+
+[[rest-monitoring-security]]
+=== Rest Monitoring Security
+
+NOTE This requires secure admin to be enabled.
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:68]: NOTE:

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+|String
+|The config whose REST Monitoring application’s configuration will be retrieved.
+|`server-config`
+|no
+|====
+
+[[rest-monitoring-usage]]
+== Usage
+
+INFO Only the READ Operations are supported currently
```
Comment (fturizo) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Logging and Monitoring/Monitoring Service/REST Monitoring/Rest Monitoring.adoc:151]: INFO:

**Review by fturizo — COMMENTED** [review]
Couple of small changes needed before merging

**Review by fturizo — APPROVED** [review]

**Pull request merged**